### PR TITLE
Feat: Added Product Listing with Filters

### PR DESCRIPTION
**Pull request by shuntzhere**

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -27,5 +27,5 @@
 
 body {
   background-color: var(--color-primary);
-  color: var(--color-white);
+  /* color: var(--color-white); */
```
Comment (rahulcs754) [inline src/App.css:30]: before push remove comment

**Review by rahulcs754 — APPROVED** [review]
The code looks clean.
just small suggestion

File: src/App.css (hunk 1 above)
Comment (shuntzhere) [reply src/App.css:30]: Thanks for the review and yes will remove that.

File: src/context/FilterCompose.jsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+};
+
+export const titleFunction = (state, products) => {
+  const { title } = state;
+  const productList = [...products];
+  if (title.length === 0) {
+    return productList;
+  } else {
+    return productList.filter((product) => title.includes(product.title));
+  }
```
Comment (nishasen) [inline src/context/FilterCompose.jsx]: filters do not modify the original array, instead it creates a new array, so you do not have to spread the array to filter it

File: src/context/FilterReducer.jsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    case "RESET":
+      return { ...defaultState };
+
+    case "TITLE":
+      const { title } = state;
+      if (title.includes(action.payload)) {
+        const updatedTitle = title.filter((t) => t !== action.payload);
+        return { ...state, title: updatedTitle };
+      } else {
+        return { ...state, title: [...state.title, action.payload] };
```
Comment (nishasen) [inline src/context/FilterReducer.jsx:19]: Do all the functionalities in compose filter function , in reducer just set the state

**Review by nishasen — APPROVED** [review]
Gave two suggestions please go through it, else the code looks clean and good.

**Pull request merged**